test/rest-coercion/urlencoded-integer.suite.js

```diff
@@ -0,0 +1,73 @@
+// Copyright IBM Corp. 2014,2016. All Rights Reserved.
+// Node module: strong-remoting
+// This file is licensed under the Artistic License 2.0.
+// License text available at https://opensource.org/licenses/Artistic-2.0
+
+'use strict';
+
+var urlEncodedContext = require('./_urlencoded.context');
+
+module.exports = function(ctx) {
+  suite('query string', urlEncodedContext(ctx, 'qs'));
+  suite('form data', urlEncodedContext(ctx, 'form'));
+};
+
+function suite(prefix, ctx) {
+  var EMPTY_QUERY = ctx.EMPTY_QUERY;
+  var ERROR_BAD_REQUEST = ctx.ERROR_BAD_REQUEST;
+  var verifyTestCases = ctx.verifyTestCases;
+
+  describe(prefix + ' - integer - required', function() {
+    // See verifyTestCases' jsdoc for details about the format of test cases.
+    verifyTestCases({ arg: 'arg', type: 'integer', required: true }, [
+      // Valid values
+      ['arg=0', 0],
+      ['arg=1', 1],
+      ['arg=-1', -1],
+
+      // Empty values should trigger ERROR_BAD_REQUEST
+      [EMPTY_QUERY, ERROR_BAD_REQUEST],
+      ['arg', 0],
+      ['arg=', 0],
+
+      // Empty-like values should trigger ERROR_BAD_REQUEST too
+      ['arg=undefined', ERROR_BAD_REQUEST],
+      ['arg=null', ERROR_BAD_REQUEST],
+    ]);
+  });
+
+  describe(prefix + ' - integer - optional', function() {
+    // See verifyTestCases' jsdoc for details about the format of test cases.
+    verifyTestCases({ arg: 'arg', type: 'integer' }, [
+      // Empty values
+      [EMPTY_QUERY, undefined],
+      ['arg', 0], // should be: undefined
+      ['arg=', 0], // should be: undefined
+
+      // Valid values
+      ['arg=0', 0],
+      ['arg=1', 1],
+      ['arg=-1', -1],
+
+      // Numbers larger than MAX_SAFE_INTEGER get trimmed
```

Comment:
Should be rejected, see https://github.com/strongloop/strong-remoting/pull/323/files/e8649ef1693273e0c8fa698fa7c7bfc70d422552..e066d0e76c608e014d312f6efac0281c08392bc0#r72065917